src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts

```diff
@@ -0,0 +1,284 @@
+      // modify the defaults
+      testComponent.secondaryActions = [
+        {
+          buttonText: 'some action',
+          isDisabled: true,
+          click: () => testComponent.secondaryActionClicked(0)
+        }
+      ];
+      fixture.detectChanges();
+      await fixture.whenStable();
+
+      const actions = summaryActionBarFixture.secondaryActions;
+
+      for (let i = 0; i < actions.length; i++) {
+        const action = actions[i];
+
+        // verify button state
+        expect(action.buttonText).toEqual(testComponent.secondaryActions[i].buttonText);
+        expect(action.isDisabled).toEqual(testComponent.secondaryActions[i].isDisabled);
+      }
+    });
+
+  });
+
+  describe('cancel button', () => {
+
+    it('should expose cancel action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
+      const action = summaryActionBarFixture.cancelAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'cancelActionClicked');
+      const action = summaryActionBarFixture.cancelAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose cancel action button properties', async () => {
+      // modify the defaults
+      testComponent.cancelAction = {
+        buttonText: 'some action',
+        isDisabled: true,
+        click: () => testComponent.cancelActionClicked()
+      };
+      fixture.detectChanges();
+      await fixture.whenStable();
+
+      const action = summaryActionBarFixture.cancelAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.cancelAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.cancelAction.isDisabled);
+    });
+  });
+
+  describe('summary content', () => {
+
+    it('should expose content body on large screen', async () => {
+      // verify the content matches
+      const summaryBody = SkyAppTestUtility.getText(summaryActionBarFixture.summaryBody);
+      expect(summaryBody).toEqual(testComponent.summaryBody);
+    });
+
+    it('should expose content body when responsive', async () => {
```

Comment:
I believe you can get the content body even when it is not visible. Should we have a test for that?